libnetwork/drivers/overlay/ov_network.go

```diff
@@ -609,7 +609,16 @@ func (n *network) setupSubnetSandbox(s *subnet, brName, vxlanName string) error
 
 	err := createVxlan(vxlanName, s.vni, n.maxMTU())
 	if err != nil {
-		return err
+		if !strings.Contains(err.Error(), "file exists") {
```

Comment:
Can we use `os.IsExists(err)` or `errors.Is(err, os.ErrExist)` https://golang.org/pkg/os/#IsExist here instead of string matching?

```suggestion
		if !errors.Is(err, os.ErrExist) {
```

looks like `createVxlan()` is not properly wrapping the error (so the underlying error gets lost), so that function would need some changing https://github.com/moby/moby/blob/a77317882d010b884a9101c6ad0b2d7db141082f/libnetwork/drivers/overlay/ov_utils.go#L74

e.g.;

```go
return errors.Wrap(err, "failed to create vxlan interface")
```

Reply:
Are we able to write a unit test for this situation? (e.g. create a vxlan interface, then create a new one with the same name, so that we force a conflict for the test)?

Reply:
Thanks! Updated it per your suggestion. I added a unit test, not sure if it is what you wanted, it needs root user to run this test (creating a vxlan interface). Is CI user a root user?

Reply:
It looks like current CI user is root, the previous unit test passed. I changed a little bit to the unit test by detecting OS and user, it is not linux or root user, the test is skipped.